File: first_set.c
#include <stdio.h>
#include <string.h>

void findFirst(char *prod) {
    printf("FIRST(%c) = { ", prod[0]);
    for (int i = 3; i < strlen(prod); i++) {
        if (prod[i] >= 'a' && prod[i] <= 'z') {
            printf("%c ", prod[i]);
            break;
        } else if (prod[i] == '|') continue;
    }
    printf("}\n");
}

int main() {
    char prod[50];
    printf("Enter production (example: E->aA|b): ");
    scanf("%s", prod);
    findFirst(prod);
    return 0;
}

/*
Example Inputs & Outputs:
1?? E->aA|b  ?  FIRST(E) = { a }
2?? S->bB|a  ?  FIRST(S) = { b }
3?? A->e|dA  ?  FIRST(A) = { e }
*/

File: first_set.java
import java.util.*;

public class FirstSet {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter production (example: E->aA|b): ");
        String prod = sc.next();

        char nt = prod.charAt(0);
        System.out.print("FIRST(" + nt + ") = { ");

        for (int i = 3; i < prod.length(); i++) {
            char ch = prod.charAt(i);
            if (ch >= 'a' && ch <= 'z') {
                System.out.print(ch + " ");
                break;
            } else if (ch == '|') continue;
        }
        System.out.println("}");
    }
}

/*
Example Inputs & Outputs:
1️⃣ E->aA|b  →  FIRST(E) = { a }
2️⃣ S->bB|a  →  FIRST(S) = { b }
3️⃣ A->ε|dA  →  FIRST(A) = { ε }
*/
